components/net/image_cache.rs

```diff
@@ -496,7 +499,21 @@ impl ImageCache for ImageCacheImpl {
         match (action, id) {
             (FetchResponseMsg::ProcessRequestBody, _) |
             (FetchResponseMsg::ProcessRequestEOF, _) => return,
-            (FetchResponseMsg::ProcessResponse(_response), _) => {}
+            (FetchResponseMsg::ProcessResponse(_response), _) => {
+                let mut store = self.store.lock().unwrap();
+                let pending_load = store.pending_loads.get_by_key_mut(&id).unwrap();
+                let metadata = match _response {
+                    Ok(meta) => {
+                        Some(match meta {
+                            FetchMetadata::Unfiltered(m) => m,
+                            FetchMetadata::Filtered { unsafe_, .. } => unsafe_,
+                        })
+                    },
+                    Err(_) => None,
+                };
+                let final_url = metadata.unwrap().final_url;
```

Comment:
This will panic if we get a network error (the `Err` case in the previous match). We should use `metadata.as_ref().map(|m| m.final_url)` to get a `Option<ServoUrl>` value.